WSLBlobNFS/wsl2_linux_script.sh

```diff
@@ -181,51 +421,51 @@ if [[ $1 == "installsystemd" ]]; then
 
 # else if the first argument is installnfssmb, then run the setup steps
 elif [[ $1 == "installnfssmb" ]]; then
+
+    if [[ $# != 2 ]]; then
+        eecho "Usage: $0 <installnfssmb> <samba username>"
+        exit 1
+    fi
+
+    sambausername=$2
+
+    apt-get update > /dev/null
     install_nfs
     install_samba
-    onetime_samba_setup $2
+    onetime_samba_setup $sambausername
 
-# else if the first argument is mountshare, then run the mount steps
+# else if the first argument is mountshare, then run the mount in nfs and share via samba
 elif [[ $1 == "mountshare" ]]; then
 
-    # Split the mount command into tokens
-    IFS=' ' read -ra nametokens <<< "$2"
-    # last element is the mount point
-    mountpoint=${nametokens[-1]}
-    # check if the last element is a path
-    if [[ ${mountpoint:0:1} == "/" ]]; then
-        # check if the path exists
-        if [[ ! -d $mountpoint ]]; then
-            mkdir -p $mountpoint
-            echo "Created $mountpoint"
-        else
-            echo "Mount point $mountpoint already exists"
-        fi
-    else
-        echo "Mount point $mountpoint is not a path"
+    if [[ $# != 4 ]]; then
+        eecho "Usage: $0 <mountshare> <mountParameterType> <mountParameter> <tempFilePath>"
         exit 1
     fi
 
-    echo "Mounting NFS share.."
+    mountparametertype=$2
+    mountparameter=$3
+    tempFilePath=$4
 
-    # quote the mount command to preserve the spaces
-    mount_nfs "$2" "$mountpoint"
-    echo "Done NFS mounting."
+    mount_share $mountparametertype $mountparameter $tempFilePath
```

Comment:
you don't seem to be using the params in mount_share.. remove these for clarity then